extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/config/OpenshiftConfig.java

```diff
@@ -0,0 +1,406 @@
+     * using the following rules:
+     */
+    @ConfigItem
+    Optional<String> version;
+
+    /**
+     * Custom labels to add to all resources
+     */
+    @ConfigItem
+    Map<String, String> labels;
+
+    /**
+     * Custom annotations to add to all resources
+     */
+    @ConfigItem
+    Map<String, String> annotations;
+
+    /**
+     * Environment variables to add to all containers
+     */
+    @ConfigItem
+    Map<String, EnvConfig> envVars;
+
+    /**
+     * Working directory
+     */
+    @ConfigItem
+    Optional<String> workingDir;
+
+    /**
+     * The commands
+     */
+    @ConfigItem
+    Optional<List<String>> command;
+
+    /**
+     * The arguments
+     *
+     * @return The arguments
+     */
+    @ConfigItem
+    Optional<List<String>> arguments;
+
+    /**
+     * The service account
+     */
+    @ConfigItem
+    Optional<String> serviceAccount;
+
+    /**
+     * The host under which the application is going to be exposed
+     */
+    @ConfigItem
+    Optional<String> host;
+
+    /**
+     * The application ports
+     */
+    @ConfigItem
+    Map<String, PortConfig> ports;
+
+    /**
+     * The type of service that will be generated for the application
+     */
+    @ConfigItem(defaultValue = "ClusterIP")
+    ServiceType serviceType;
+
+    /**
+     * Image pull policy
+     */
+    @ConfigItem(defaultValue = "IfNotPresent")
+    ImagePullPolicy imagePullPolicy;
+
+    /**
+     * The image pull secret
+     */
+    @ConfigItem
+    Optional<List<String>> imagePullSecrets;
+
+    /**
+     * The liveness probe
+     */
+    @ConfigItem
+    Optional<ProbeConfig> livenessProbe;
+
+    /**
+     * The readiness probe
+     */
+    @ConfigItem
+    Optional<ProbeConfig> readinessProbe;
+
+    /**
+     * Volume mounts
+     */
+    @ConfigItem
+    Map<String, MountConfig> mounts;
+
+    /**
+     * Secret volumes
+     */
+    @ConfigItem
+    Map<String, SecretVolumeConfig> secretVolumes;
+
+    /**
+     * ConfigMap volumes
+     */
+    @ConfigItem
+    Map<String, ConfigMapVolumeConfig> configMapVolumes;
+
+    /**
+     * Git Repository volumes
+     */
+    @ConfigItem
+    Map<String, GitRepoVolumeConfig> gitRepoVolumes;
+
+    /**
+     * PVC volumes
+     */
+    @ConfigItem
+    Map<String, PvcVolumeConfig> pvcVolumes;
+
+    /**
+     * AWS Elastic BlockStore volumes
+     */
+    @ConfigItem
+    Map<String, AwsElasticBlockStoreVolumeConfig> awsElasticBlockStoreVolumes;
+
+    /**
+     * Azure file volumes
+     */
+    @ConfigItem
+    Map<String, AzureFileVolumeConfig> azureFileVolumes;
+
+    /**
+     * Azure disk volumes
+     */
+    @ConfigItem
+    Map<String, AzureDiskVolumeConfig> azureDiskVolumes;
+
+    /**
+     * Init containers
+     */
+    @ConfigItem
+    Map<String, ContainerConfig> initContainers;
+
+    /**
+     * Sidecar containers
+     */
+    @ConfigItem
+    Map<String, ContainerConfig> containers;
+
+    public Optional<String> getGroup() {
```

Comment:
Let's please get rid of all the getters and setters in all the configuration objects that were added.